packages/store/src/onboardingSlice/index.ts

```diff
@@ -11,11 +11,24 @@
 //    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 //    See the License for the specific language governing permissions and
 //    limitations under the License.
-import { createSlice } from '@reduxjs/toolkit';
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
 
 export const sliceName = 'onboarding';
 
-export const initialState = {
+export type OnboardingApproach = 'virtual' | 'physical';
+type OnboardingSliceType = {
+  address?: string;
+  approach: OnboardingApproach | null;
+  complete: boolean;
+  demoArtifactPort: number;
+  deviceConnection?: string;
+  deviceType: string[] | null;
```

Comment:
```suggestion
  deviceType: string | string[] | null;
```
due to the devices potentially sending a `string | string[]`, but in most cases it's a `string` + adjusting the `[<deviceType>]` in other places